sw/device/lib/drivers/power_manager/power_manager.c

```diff
@@ -15,19 +15,20 @@
 
 void power_gate_core_asm()
 {
-
     asm volatile (
 
         // write POWER_GATE_CORE[0] = 1
         "lui a0, %[base_address_20bit]\n"
-        "li  a1, 1\n"
-        "sw  a1, %[power_manager_power_gate_core_reg_offset](a0)\n"
+        "lw  a1, %[power_manager_power_gate_core_reg_offset](a0)\n"
+        "li  a2, 1\n"
```

Comment:
the SW does not know that the CPU is the bit 0, so you should have added a bit field for each power domain, and use the bitwise operations to do these computations / or, just add a register with 1 bit for each power domain

Reply:
Solved.